internal-services/catalog/publish-index-image-task.yaml

```diff
@@ -49,14 +49,46 @@ spec:
         quay.io/konflux-ci/release-service-utils:e633d51cd41d73e4b3310face21bb980af7a662f
       script: |
         #!/usr/bin/env bash
+        set -euo pipefail
+
         PATH=/bin:/usr/bin:/usr/local/bin
         export PATH
 
-        # do not authenticate if the source is redhat's "registry-proxy" which is unauthenticated.
+        AUTH_PARAM=()
+        # Do not authenticate if the source is Red Hat's "registry-proxy" which is unauthenticated.
         if [[ ! "$(params.sourceIndex)" =~ ^registry-proxy(\-stage)?.engineering.redhat.com ]]; then
             AUTH_PARAM=("--src-creds" "${SOURCE_INDEX_CREDENTIAL}")
         fi
+        TARGET_AUTH="--dest-creds ${TARGET_INDEX_CREDENTIAL}"
+
+        echo "Getting source image digest: $(params.sourceIndex)"
+        SOURCE_DIGEST=$(skopeo inspect \
+            --tls-verify=false \
+            "${AUTH_PARAM[@]}" \
+            "docker://$(params.sourceIndex)" \
+            --format '{{.Digest}}')
+
+        echo "Getting target image digest: $(params.targetIndex)"
+        if TARGET_DIGEST=$(skopeo inspect \
+            --tls-verify=false \
+            "${TARGET_AUTH}" \
```

Comment:
I don't think this will work. This would result in one cli argument `"--dest-creds ${TARGET_INDEX_CREDENTIAL}"` - you want two. So you should use an array - same way `AUTH_PARAM` is done. And since you already create this, I would suggest to also use the array in the `skopeo copy` command at the end.

Reply:
Also, I would suggest better names for the variables: `SOURCE_AUTH_ARGS` and `TARGET_AUTH_ARGS`.